docs/delivery_attempts.rst

```diff
@@ -0,0 +1,253 @@
+   ...             to="https://example.com/some/path"
+   ...             for="zope.container.interfaces.IContentContainer"
+   ...             when="zope.lifecycleevent.interfaces.IObjectCreatedEvent" />
+   ... </configure>
+   ... """)
+
+Now we can access the subscription. Because delivery attempts are
+stored in the subscription, it has a length of 0 at this time.
+
+.. doctest::
+
+
+   >>> from zope import component
+   >>> from nti.webhooks import interfaces
+   >>> sub_manager = component.getUtility(interfaces.IWebhookSubscriptionManager)
+   >>> subscription = sub_manager['Subscription']
+   >>> len(subscription)
+   0
+
+To avoid actually trying to talk to example.com, we'll be using some mocks.
+
+.. doctest::
+
+   >>> from nti.webhooks.testing import mock_delivery_to
+   >>> mock_delivery_to('https://example.com/some/path', method='POST', status=200)
+
+Now we will create the object, broadcast the event to engage the
+subscription, and commit the transaction to send the hook.
+
+.. doctest::
+
+   >>> import transaction
+   >>> from zope import lifecycleevent
+   >>> from zope.container.folder import Folder
+   >>> _ = transaction.begin()
+   >>> lifecycleevent.created(Folder())
+   >>> transaction.commit()
+
+In the background, the `IWebhookDeliveryManager` is busy invoking the hook. We need to wait for it to
+finish, and then we can examine our delivery attempt:
+
+.. doctest::
+
+   >>> from zope import component
+   >>> from nti.webhooks.interfaces import IWebhookDeliveryManager
+   >>> component.getUtility(IWebhookDeliveryManager).waitForPendingDeliveries()
+
+Attempt Details
+~~~~~~~~~~~~~~~
+
+The subscription now has an attempt recorded in the form of an
+``IWebhookDeliveryAttempt``. The attempt records some basic details,
+such as the overall status, and an associated message.
+
+.. important::
+
+   Attempts are immutable. They are created and managed entirely by
+   the system and mutation attempts are not allowed.
+
+
+.. doctest::
+
+   >>> len(subscription)
+   1
+   >>> attempt = subscription.pop()
+   >>> from zope.interface import verify
+   >>> verify.verifyObject(interfaces.IWebhookDeliveryAttempt, attempt)
+   True
+   >>> attempt.status
+   'successful'
+   >>> print(attempt.message)
+   200 OK
+
+An important attribute of the attempt is the ``request``; this
+attribute (an `IWebhookDeliveryAttemptRequest`) provides information
+about the HTTP request as it went on the wire.
+
+.. doctest::
+
+   >>> verify.verifyObject(interfaces.IWebhookDeliveryAttemptRequest, attempt.request)
+   True
+   >>> print(attempt.request.url)
+   https://example.com/some/path
+   >>> print(attempt.request.method)
+   POST
+   >>> import pprint
+   >>> pprint.pprint({str(k): str(v) for k, v in attempt.request.headers.items()})
+   {'Accept': '*/*',
+    'Accept-Encoding': 'gzip, deflate',
+    'Connection': 'keep-alive',
+    'Content-Length': '94',
+    'Content-Type': 'application/json',
+    'User-Agent': 'nti.webhooks ...'}
```

Comment:
Should we be obscuring or popping off any headers that the dialect could be placing secrets in? i.e. Authorization?

Reply:
That's [the second time you've asked that](https://github.com/NextThought/nti.webhooks/pull/2#discussion_r450574564) 😉

Reply:
I thought it was just deja-vu 😄. carry on